8932 container base image (#8933)
IQSS/dataverse: modules/container-base/README.md

> # Dataverse Base Container Image

Reviewer: I'm surprised that it reads "Dataverse Base Container Image" here but "Application Base Image" on the [rendered version](https://dataverse-guide--8933.org.readthedocs.build/en/8933/container/base-image.html). But as far as the code here is concerned, shouldn't it be "Dataverse Container Base Image"?

Author: Beware - this text is not rendered within the guide, but at https://hub.docker.com/r/gdcc/base. I would like the Hub page to be much shorter and self explaining. Do you think we should align both?

Reviewer: I thought the last link you provided was a more reader-friendly version of the code. But then I saw several sections weren't on GitHub. Either way, I was just curious about the phrasing of this specific section. Nothing vital here  :thumbsup:

Author: OK closing then. Thanks again! :smile:

> A "base image" offers you a pre-installed and pre-tuned application server to deploy Dataverse software to.

Author: Thanks for the review @atrisovic!

I like your idea, but I'm afraid it would not be what is happening here. The resulting image is not containing a single bit of Dataverse software or scripts yet, which I would read from your suggestion. But I see what you are getting at!

> A "base image" offers you a pre-installed and pre-tuned application server to deploy Dataverse software to.

Reviewer 2: Thanks for the prompt reply @poikilotherm ! :)
How about:

The Dataverse Base Container Image contains pre-installed and pre-tuned application server that contains the necessary software dependencies for deploying and launching a Dataverse repository installation, but it does not contain the Dataverse software itself.

> A "base image" offers you a pre-installed and pre-tuned application server to deploy Dataverse software to.

Author: Like that?

```suggestion
The Dataverse Base Container Image contains primarily a pre-installed and pre-tuned application server. That contains the necessary software dependencies for deploying and launching a Dataverse repository installation, but it does not contain the Dataverse software itself.

```

> A "base image" offers you a pre-installed and pre-tuned application server to deploy Dataverse software to.

Reviewer 2: Yeah I think so, if it is correct 😄 

I would keep it as a single sentence (but it's just style): 
The Dataverse Base Container Image contains primarily a pre-installed and pre-tuned application server with the necessary software dependencies for deploying and launching a Dataverse repository installation.

Maybe then at the end of the paragraph: The image does not contain the Dataverse software itself.

> A "base image" offers you a pre-installed and pre-tuned application server to deploy Dataverse software to.

Author: :heavy_check_mark:

> Adding basic functionality like executing scripts at container boot, monitoring, memory tweaks etc is all done

Reviewer: ```suggestion
Adding basic functionality like executing scripts at container boot, monitoring, memory tweaks, etc., is all done
```

> at this layer, to make the application image focus on the app itself.

Reviewer 3: Perhaps we should say that the application image is coming or planned. Make it clear that it's not yet available, I mean. Maybe we could link to an open issue.

Author: :heavy_check_mark:

## Quick Reference

**Maintained by:** 

This image is created, maintained and supported by the Dataverse community on a best-effort basis.

**Where to find documentation:**

The [Dataverse Container Guide - Base Image](https://guides.dataverse.org/en/latest/container/base-image.html)
provides in-depth information about content, building, tuning and so on for this image. 

**Where to get help and ask questions:**

> IQSS will not offer you support how to deploy or run it, please reach out to the community for help on using it.

Reviewer: ```suggestion
IQSS will not offer you support how to deploy or run it. Please reach out to the community for help on using it.
```

> IQSS will not offer you support how to deploy or run it. Please reach out to the community for help on using it.

Reviewer 3: Better would be "support on how to..."

Author: :heavy_check_mark:

You can join the Community Chat on Matrix at https://chat.dataverse.org or the Community Slack at
https://dataversecommunity.slack.com to ask for help and guidance.

## Supported Image Tags

This image is sourced within the main upstream code [repository of the Dataverse software](https://github.com/IQSS/dataverse).

> Development and maintenance happens there (again, by the community). Community supported image tags are based on the two

Reviewer: ```suggestion
Development and maintenance happens there (again, by the community). Community-supported image tags are based on the two
```

> Development and maintenance happens there (again, by the community). Community-supported image tags are based on the two

Reviewer 3: By now the reader is quite clear who is supporting this thing... the community! 😄 I'm ok with this fact being over-communicated. 😄

most important branches:

- `develop` representing the unstable state of affairs in Dataverse's development branch
  ([`Dockerfile`](https://github.com/IQSS/dataverse/tree/develop/modules/container-base/src/main/docker/Dockerfile))

> - `release` representing the latest stable release in Dataverse's main branch

Reviewer 3: You've lost me. Shouldn't this be `master` rather than `release`?

Reviewer 3: Oh, you're talking about tags here. Hmm. I find it confusing that one of the tags matches a branch ("develop") and another tag does not ("release" vs "master"). I would find it less confusing if neither tag matched. For tags, we could try "stable" and "unstable" for example.

Something like this maybe:

---

This image is sourced from the main upstream code [repository of the Dataverse software](https://github.com/IQSS/dataverse). Development and maintenance of the [image code](https://github.com/IQSS/dataverse/tree/develop/modules/container-base/src/main/docker/Dockerfile) happens there (again, by the community). Community-supported image tags are based on the two most important upstream branches:

- The `unstable` tag corresponds to the `develop` branch, where pull requests are merged.
-  The`stable` tag corresponds to the `master` branch, where releases are cut from.

Author: Alright, will do!

Author: Applied as requested.

Reviewer 3: From a quick glance, looks good! Thanks!

  ([`Dockerfile`](https://github.com/IQSS/dataverse/tree/master/modules/container-base/src/main/docker/Dockerfile))

> Within the main repository, you may find the base image's files at `<git root>/modules/container-base`.

Reviewer: ```suggestion
Within the main repository, you may find the base image files at `<git root>/modules/container-base`.
```

> This Maven module uses the `Maven Docker Plugin <https://dmp.fabric8.io>`_ to build and ship the image.

Reviewer 3: This seems to be an .rst link but we should use the .md style. That is, it's not a link at https://hub.docker.com/r/gdcc/base

You may use, extend, or alter this image to your liking and/or host in some different registry if you want to.

**Supported architectures:** This image is created as a "multi-arch image", supporting the most common architectures 
Dataverse usually runs on: AMD64 (Windows/Linux/...) and ARM64 (Apple M1/M2).

## License

Image content created by the community is licensed under [Apache License, Version 2.0](https://www.apache.org/licenses/LICENSE-2.0), 
like the [main Dataverse project](https://github.com/IQSS/dataverse/blob/develop/LICENSE.md).

Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an
"AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. 
See the License for the specific language governing permissions and limitations under the License.

As with all Docker images, all images likely also contain other software which may be under other licenses (such as 

> [Payara Server](https://github.com/payara/Payara/blob/master/LICENSE.txt), Bash, etc from the base

Reviewer: ```suggestion
[Payara Server](https://github.com/payara/Payara/blob/master/LICENSE.txt), Bash, etc., from the base
```
